### PR TITLE
Mouse colon

### PR DESCRIPTION
**Pull request by mlin865**

### CODE REVIEW

File: scaffoldmaker/meshtypes/meshtype_3d_colonsegmentsimplemesentery1.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        refineElementsCountThroughWall = options['Refine number of elements through wall']
+
+        baseRegion = region.createRegion()
+        baseAnnotationGroups = cls.generateBaseMesh(baseRegion, options)
+
+        meshrefinement = MeshRefinement(baseRegion, region, baseAnnotationGroups)
+        meshrefinement.refineAllElementsCubeStandard3d(refineElementsCountAround, refineElementsCountAlong, refineElementsCountThroughWall)
+        return meshrefinement.getAnnotationGroups()
+
+def getColonSegmentInnerPoints0TC(region, elementsCountAroundMZ, elementsCountAroundNonMZ, elementsCountAlongSegment,
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colonsegmentsimplemesentery1.py]: Name wasn't immediately obvious, rename:
0TC -> NoTeniaColi, or SimpleMesentery?

File: scaffoldmaker/meshtypes/meshtype_3d_colon1.py [hunk 2]
```diff
@@ -83,47 +84,57 @@ def getParameterSetNames():
...
+            'Refine number of elements around',
+            'Refine number of elements along',
+            'Refine number of elements through wall' ]
 
     @classmethod
     def getOptionValidScaffoldTypes(cls, optionName):
         if optionName == 'Central path':
             return [ MeshType_1d_path1 ]
+        if optionName == 'Segment profile':
+            return[ MeshType_3d_colonsegmentsimplemesentery1,
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colon1.py]: Minor. Insert space after return.

File: scaffoldmaker/meshtypes/meshtype_3d_colon1.py [hunk 3]
```diff
@@ -157,23 +178,34 @@ def generateBaseMesh(region, options):
...
-        elementsCountAround = (elementsCountAroundTC + elementsCountAroundHaustrum)*3
-        elementsCountAlongHaustrum = options['Number of elements along haustrum']
-        elementsCountThroughWall = options['Number of elements through wall']
-        haustraSegmentCount = options['Number of haustra segments']
-        radius = options['Inner radius']
-        cornerInnerRadiusFactor = options['Corner inner radius factor']
-        haustrumInnerRadiusFactor = options['Haustrum inner radius factor']
-        haustrumLengthEndDerivativeFactor = options['Haustrum length end derivative factor']
-        haustrumLengthMidDerivativeFactor = options['Haustrum length mid derivative factor']
-        widthTC = options['Tenia coli width']
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colon1.py]: Make consistent with other variables, where width, thickness is last, also with naming conventions I use in heart (including variables starting with lower case, types with upper case):
tcWidth
tcThickness

File: scaffoldmaker/meshtypes/meshtype_3d_colonsegmentsimplemesentery1.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    :return annotationGroups, annotationArray: annotationArray stores annotation
+    names of elements around
+    :return transitElementList: stores true if element around is an element that
+    transits from tenia coli / mesenteric zone to haustrum / non-mesenteric zone.
+    :return uList: List of xi for node around
+    : return totalArcLengthOuterMidLength: total arclength of elements on outer
+    surface along mid-length of segment.
+    :return coordinates, derivatives on inner surface of a colon segment.
+    """
+    MZGroup = AnnotationGroup(region, 'mesenteric zone', FMANumber = 'FMANumber unknown', lyphID = 'Lyph ID unknown')
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colonsegmentsimplemesentery1.py]: Use standard of variables starting with lower case.

File: scaffoldmaker/meshtypes/meshtype_3d_colonsegmentteniacoli1.py [hunk 5]
```diff
@@ -195,17 +205,17 @@ def generateMesh(cls, region, options):
         meshrefinement.refineAllElementsCubeStandard3d(refineElementsCountAround, refineElementsCountAlong, refineElementsCountThroughWall)
         return meshrefinement.getAnnotationGroups()
 
-def getColonHaustraSegmentInnerPoints(elementsCountAroundTC, elementsCountAroundHaustrum, elementsCountAlongHaustrum, widthTC, radius,
-    cornerInnerRadiusFactor, haustrumInnerRadiusFactor, haustrumLengthEndDerivativeFactor, haustrumLengthMidDerivativeFactor, haustrumLength):
+def getColonSegmentInnerPoints3TC(region, elementsCountAroundTC, elementsCountAroundHaustrum, elementsCountAlongSegment, widthTC, radius,
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colonsegmentteniacoli1.py]: 3TeniaColi, or is it going to take number of tenia coli as an argument once you do pig? Can always:
<li>
assert teniaColiCount == 3, 'function name Only implemented for 3 tenia coli'
</li>

**Comment by mlin865** [comment]
Changes made as requested.

**Pull request merged**